lib/metadata/models/object/ObjectMD.js

```diff
@@ -0,0 +1,613 @@
+     */
+    setIsNull(isNull) {
+        this._data.isNull = isNull;
+        return this;
+    }
+
+    /**
+     * Get metadata isNull value
+     *
+     * @return {boolean} Whether new version is null or not
+     */
+    getIsNull() {
+        return this._data.isNull;
+    }
+
+    /**
+     * Set metadata nullVersionId value
+     *
+     * @param {string} nullVersionId - The version id of the null version
+     * @return {ObjectMD} itself
+     */
+    setNullVersionId(nullVersionId) {
+        this._data.nullVersionId = nullVersionId;
+        return this;
+    }
+
+    /**
+     * Get metadata nullVersionId value
+     *
+     * @return {string} The version id of the null version
+     */
+    getNullVersionId() {
+        return this._data.nullVersionId;
+    }
+
+    /**
+     * Set metadata isDeleteMarker value
+     *
+     * @param {boolean} isDeleteMarker - Whether object is a delete marker
+     * @return {ObjectMD} itself
+     */
+    setIsDeleteMarker(isDeleteMarker) {
+        this._data.isDeleteMarker = isDeleteMarker;
+        return this;
+    }
+
+    /**
+     * Get metadata isDeleteMarker value
+     *
+     * @return {boolean} Whether object is a delete marker
+     */
+    getIsDeleteMarker() {
+        return this._data.isDeleteMarker;
+    }
+
+    /**
+     * Set metadata versionId value
+     *
+     * @param {string} versionId - The object versionId
+     * @return {ObjectMD} itself
+     */
+    setVersionId(versionId) {
+        this._data.versionId = versionId;
+        return this;
+    }
+
+    /**
+     * Get metadata versionId value
+     *
+     * @return {boolean} The object versionId
+     */
+    getVersionId() {
+        return this._data.versionId;
+    }
+
+    /**
+     * Set tags
+     *
+     * @param {object} tags - tags object
+     * @return {ObjectMD} itself
+     */
+    setTags(tags) {
+        this._data.tags = tags;
+        return this;
+    }
+
+    /**
+     * Returns tags
+     *
+     * @return {object} tags object
+     */
+    getTags() {
+        return this._data.tags;
+    }
+
+    /**
+     * Set custom meta headers
+     *
+     * @param {object} data - Meta headers
+     * @return {ObjectMD} itself
+     */
+    setUserMetadata(data) {
+        Object.keys(data).forEach(key => {
+            if (key.startsWith('x-amz-meta')) {
+                this._data[key] = data[key];
+            }
+        });
+        // If a multipart object and the acl is already parsed, we update it
+        if (data.acl) {
+            this.setAcl(data.acl);
+        }
+        return this;
+    }
+
+    /**
+     * Set custom headers
+     *
+     * @param {object} data - Headers
+     * @return {ObjectMD} itself
+     */
+    setHeaders(data) {
+        Object.keys(data).forEach(key => {
+            this._data[key] = data[key];
+        });
+        return this;
+    }
+
+    /**
+     * Returns metadata object
+     *
+     * @return {object} metadata object
+     */
+    getData() {
```

Comment:
I find the name `getData` confusing, it can be taken for the actual object data. Could be name `getValue` instead for example, or `asJSObject` to point the fact that we're returning the native JS object.